### PR TITLE
Update LIS user guide

### PR DESCRIPTION
**Pull request by ErlendHaa**
Several new features and interfaces have been added since the last
update of the user guide. This commits adds relevant sections of some of
the key concepts.

### CODE REVIEW

File: python/docs/lis/userguide.rst [hunk 1]
```diff
@@ -56,26 +65,180 @@ also be reached directly from the Logical File:
...
+    60
+    >>> format_spec.spacing_units
+    '.1IN'
+    >>> format_spec.direction
+    255
+
+This tells us that the current logset is indexed against depth, and that the
+depth is measured in 1/10 of an inch. There is also a constant spacing of 60
+.1IN (or 6 IN) between each sample. Note that there is no general requirement
+that the index is evenly spaced and these values might be missing in such
```
Comment (achaikou) [inline python/docs/lis/userguide.rst:114]: these values

what values? Remove "these"?

File: python/docs/lis/userguide.rst [hunk 2]
```diff
@@ -56,26 +65,180 @@ also be reached directly from the Logical File:
...
+direction and information about the index. In addition it contains a list of
+Spec Blocks, :attr:`dlisio.lis.DataFormatSpec.specs`. Each Spec Block describes
+one of the channels/curves in the logset. But let's begin by looking at the
+index:
+
+.. code-block:: python
+
+    >>> format_spec = formatspecs[0]
+    >>> format_spec.index_mnem
+    'TDEP'
```
Comment (achaikou) [inline python/docs/lis/userguide.rst:101]: Maybe use DEPT here and below?
I guess you used some real file, but according to customer Subset, DEPT is dictionary-controlled value, while TDEP is not there.

(You also have inconsistencies in your testimony and use DEPT sometimes already :) )

File: python/docs/lis/userguide.rst [hunk 3]
```diff
@@ -56,26 +65,180 @@ also be reached directly from the Logical File:
...
+logset. To read the actual curves, pass the DFSR ``format_spec`` to
+:func:`dlisio.lis.curves` together with the filehandle ``f``:
+
+.. code-block:: python
+
+    >>> curves = lis.curves(f, format_spec)
+
+This returns a `structured numpy.ndarray`_ containing all the curves in the
+logset, using the mnemonics from the Spec Blocks as field (column) names.
+
```
Comment (achaikou) [inline python/docs/lis/userguide.rst:148]: This returns a ... . Array contains all the curves in the logset and uses the mnemonics...

"using" now refers to wrong word

File: python/docs/lis/userguide.rst [hunk 4]
```diff
@@ -56,26 +65,180 @@ also be reached directly from the Logical File:
...
+Fast Channels
+.............
+
+There is one caveat to reading the curves from a logset that is necessary to be
+aware of. LIS supports a concept it refers to as 'Fast Channels'. In short this
+means that channels within the *same* logset can have different sampling rates.
+
+When a logset contains curves with different sampling rates, these have to be
+read separately. I.e. multiple calls to the :func:`dlisio.lis.curves` is
```
Comment (achaikou) [inline python/docs/lis/userguide.rst:157]: multiple calls ... (are) necessary

File: python/docs/lis/userguide.rst [hunk 5]
```diff
@@ -56,26 +65,180 @@ also be reached directly from the Logical File:
...
+that ``RCCL`` is sampled once every inch.
+
+You can also get a set of all sample rates in the DataFormatSpec:
+
+.. code-block:: python
+
+    >>> format_spec.sample_rates()
+    {1, 6, 30}
+
+There are 3 different sampling rates, which be must read with separate calls to
```
Comment (achaikou) [inline python/docs/lis/userguide.rst:181]: which (must be) read

And maybe not read, but "which must be used in separate calls to ...". After all, you are are not reading the rates, you read the data.

File: python/docs/lis/userguide.rst [hunk 6]
```diff
@@ -56,26 +65,180 @@ also be reached directly from the Logical File:
...
+    >>> data_06 = lis.curves(f, format_spec, sample_rate=6)
+    >>> data_30 = lis.curves(f, format_spec, sample_rate=30)
+
+.. note::
+    Please refer to :func:`dlisio.lis.curves` for a more comprehensive
+    explanation of this topic.
+
+.. note::
+    When reading channels with a higher sampling rate, the index is still
+    included in the resulting numpy array, and the missing values have been
```
Comment (achaikou) [inline python/docs/lis/userguide.rst:196]: the missing vales (are) linearly interpolated.

(time consistency with "index is included".)

File: python/docs/lis/userguide.rst [hunk 7]
```diff
@@ -56,26 +65,180 @@ also be reached directly from the Logical File:
...
+.. code-block:: python
+
+    >>> data = lis.curves(f, format_spec, sample_rate=30)
+    >>> meta = lis.curves_metadata(format_spec, sample_rate=30)
+
+When passing the same DFSR and ``sample_rate`` to
+:func:`dlisio.lis.curves_metadata` and :func:`dlisio.lis.curves` you get a
+dict with the corresponding Spec Blocks. Like the numpy array returned by
+``curves()`` this dict uses the mnemonics from the Spec Blocks as keys. This
+makes it easy to get both the curve itself and it's metadata:
```
Comment (achaikou) [inline python/docs/lis/userguide.rst:216]: and (its) metadata

File: python/docs/lis/userguide.rst [hunk 8]
```diff
@@ -56,26 +65,180 @@ also be reached directly from the Logical File:
...
+.............................
+
+:func:`dlisio.lis.curves` has a sister function
+:func:`dlisio.lis.curves_metadata`. This function can be useful if you want to
+associate the curves and metadata (Spec Blocks):
+
+.. code-block:: python
+
+    >>> data = lis.curves(f, format_spec, sample_rate=30)
+    >>> meta = lis.curves_metadata(format_spec, sample_rate=30)
```
Comment (achaikou) [inline python/docs/lis/userguide.rst:210]: Maybe explicitly note that this doesn't need "f" reference? By other similarities it looks like you missed it here.

Comment (ErlendHaa) [reply python/docs/lis/userguide.rst:210]: Neh

**Review by achaikou — APPROVED** [review]

**Pull request merged**